### ISSUE TITLE
Client-side muting of specific users

### ISSUE THREAD

**Issue by quazgar** [bigbluebutton/bigbluebutton#10643]
**Problem: Two participants in same room**
When two participants in a meeting are physically in the same room (different computers), they will hear the other person twice: once directly and once with a lag via BBB.

**Proposed solution: client-specific muting of participants**
- Person B can decide to mute and unmute person A for themself.
  - Person B will thus hear person A only directly "over air", not via BBB.
  - Person C (at some other location) is still able to hear person A as before.

**Alternatives**
None, unless the lag can be decreased to really small values for participants in a local network.

**Comment by quazgar** [comment]
Note: Jitsi has this feature enabled, this is why I wished it would exist in BBB as well.

**Comment by quazgar** [comment]
Any yes/no for this enhancement request?

**Comment by ffdixon** [comment]
Moderators can mute other users -- and BigBlueButton makes it easy to see who is generating audio with their name highlighted name appearing above the presentation window.  As a moderator, you can click on the highlighted name as a shortcut to mute that user.

Is that sufficient for your use case?  By default, viewers (students) can't mute each other.

**Comment by quazgar** [comment]
Not exactly, the scenario is that as an unprivileged user, I want to be able to mute the audio stream from one specific user, just for me. The use case is that when user A and B are sitting in the same physical room, having a BBB chat with some conference, they don't want to hear each other double, physically *and* over BBB.
![bbb_mute](https://user-images.githubusercontent.com/302587/148701099-2e288615-4ab3-4535-a57a-35c8b63ae662.png)

**Comment by ticxx** [comment]
We also need this feature. For hybrid learning, where some students are in the classroom and some students participate from home.

**Comment by quazgar** [comment]
What could be done to help on this issue?

**Comment by antobinary** [comment]
Hi @quazgar 

For the case where there A and B are in the same room, I often find myself pressing :mute: on my keyboard while B is talking, and also keeping an eye on the who-is-talking area above the presentation area, so that I can unmute in case C from another room starts talking over B... Alternatively sometimes I just pull my headset away a bit, or lower the volume, just so that I can be in position to quickly hop back into the full immersion experience.

I see that there are a number of upvoters here but have to say that this is not currently on our near-term roadmap. There is a level of complexity involved here -- for example if B is talking, I think it would be best if A "unsubscribes" from audio from B, however that touches on mixing of audio channels.

**Comment by quazgar** [comment]
> This issue has been automatically marked as stale because it has not had recent activity. It will be closed if no further activity occurs. Thank you for your contributions.

Ping

**Comment by Fladenbrot** [comment]
I need this feature so much

**Comment by quazgar** [comment]
Ping

**Comment by n8falke** [comment]
It is probably not possible solve this client side. The audio is mixed in freeswitch.

In our setup we had the same problem and solved it with an admin page allowing to configure freeswitch settings like:

- relate mute specific connections (https://developer.signalwire.com/freeswitch/FreeSWITCH-Explained/Modules/mod_conference_3965534/#api-reference)
- volume_in
- volume_out
- energy (level for surround noise muting)

**Comment by gustavotrott** [comment]
> It is probably not possible solve this client side. The audio is mixed in freeswitch.
> 
> In our setup we had the same problem and solved it with an admin page allowing to configure freeswitch settings like:
> 
> * relate mute specific connections (https://developer.signalwire.com/freeswitch/FreeSWITCH-Explained/Modules/mod_conference_3965534/#api-reference)
> * volume_in
> * volume_out
> * energy (level for surround noise muting)

With LiveKit becoming the default, this may be technically possible now, since audio is handled as individual participant tracks.
Is that correct, @prlanzarin ?

From 4.0 LiveKit will be default: https://github.com/bigbluebutton/bigbluebutton/pull/25239

**Comment by prlanzarin** [comment]
Correct. Per user volume control and selective subscription/muting (which will likely end up having the same core primitives) is something for 4.0+, LiveKit only.
How and when that will be done needs to be discussed.
